aws/logs_monitoring/lambda_function.py

```diff
@@ -822,13 +846,6 @@ def awslogs_handler(event, context, metadata):
                 # Add the arn as a log attribute
                 arn_attributes = {"lambda": {"arn": arn}}
                 aws_attributes = merge_dicts(aws_attributes, arn_attributes)
-                # Add the function name as tag
```

Comment:
If I understand correctly, `metadata` is supposed to only carry metadata that applies to **all** the log events, and the root issue is using `metadata` to carry event-specific tags, such as `functionname`?  

If so, I would apply the functionname, host and service to the log event right here, instead of adding `add_metadata_to_lambda_log` to the `split`. We can either loop through `logs["logEvents"]` right here and set these attributes, or cache the attributes in something and apply it later in the existing for loop, like `aws_attributes`.